src/db/migration/V1_1710848004576__Palautteet_skeema.sql

```diff
@@ -0,0 +1,101 @@
+CREATE TYPE heratelahteet AS ENUM (
+  'ehoks_update',
+  'koski_update',
+  'niputus'
+);
+
+CREATE TYPE kyselytyypit AS ENUM (
+  'aloittaneet',
+  'valmistuneet',
+  'osia_suorittaneet',
+  'tyopaikkajakson_suorittaneet',
+  'tpo_nippu',
+  'tpk_nippu'
+);
+
+CREATE TYPE palautetilat AS ENUM (
+  'ei_laheteta',
+  'kysely_muodostettu',
+  'lahetetty',
+  'lahetys_epaonnistunut',
+  'niputettu',
+  'odottaa_kasittelya',
+  'tpk_niputettu',
+  'vastaajatunnus_muodostettu',
+  'vastattu',
+  'vastausaika_loppunut'
+);
+
+CREATE TABLE IF NOT EXISTS palautteet (
+  id SERIAL PRIMARY KEY,
+  created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+  updated_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+  deleted_at TIMESTAMP WITH TIME ZONE,
+  herate_source heratelahteet NOT NULL,
+  heratepvm DATE,
+  kyselytyyppi kyselytyypit NOT NULL,
+  tila palautetilat NOT NULL,
+  kyselylinkki VARCHAR(64),
+  arvo_tunniste TEXT,
+  voimassa_alkupvm DATE,
+  voimassa_loppupvm DATE,
+  jakson_yksiloiva_tunniste VARCHAR(256),
+  hoks_id INTEGER NOT NULL,
+  nippu_id INTEGER,
+  koulutustoimija TEXT,
+  hankintakoulutuksen_toteuttaja TEXT,
+  toimipiste_oid TEXT,
+  suorituskieli CHAR(2),
+  tutkintotunnus INTEGER,
+  tutkintonimike TEXT,
+  FOREIGN KEY (hoks_id) REFERENCES hoksit (id) ON DELETE CASCADE,
+  FOREIGN KEY (nippu_id) REFERENCES palautteet (id) ON DELETE CASCADE
```

Comment:
Koska palautteet ja HOKSit on erillisiä, tän kannattais ehkä olla ON DELETE NO ACTION tai ON DELETE SET NULL (eka ois ehkä parempi, eli ei tueta hard deleteä ollenkaan HOKSille, josta on muodostettu palaute).  Mitä tulee nippuihin, periaatteessa työelämäpalautteen pystyis niputtamaan uudelleen vaikka nippu häippäisikin joten sekin vois olla ON DELETE SET NULL.

Reply:
Hyviä huomioita! Joo muutin palautteet-taulun delete actionit juurikin näin.